### PR TITLE
`S3Path` make `config` field last

### PR DESCRIPTION
**Pull request by omus**
Follow up on aside mentioned in: https://github.com/JuliaCloud/AWSS3.jl/pull/182#discussion_r687347332. I want to get this into the 0.8 release so the deprecations can be dropped as part of 0.9

### CODE REVIEW

**Comment by omus** [comment]
bors try

File: test/s3path.jl [hunk 1]
```diff
@@ -420,41 +420,41 @@ function s3path_tests(config)
         cfg = global_aws_config()
 
         @test S3Path("s3://my_bucket/prefix/that/is/fun") == S3Path(
-            ("prefix", "that", "is", "fun"), "/", "s3://my_bucket", false, cfg, nothing
+            ("prefix", "that", "is", "fun"), "/", "s3://my_bucket", false, nothing, cfg,
```
[the remark this answers is not in the document]
Comment (omus) [reply test/s3path.jl:423]: I don't think this is correct but I'll go with it for now: https://github.com/domluna/JuliaFormatter.jl/issues/455

File: src/AWSS3.jl [hunk 2]
```diff
@@ -36,6 +36,7 @@ export S3Path,
     s3_delete_tags
 
 using AWS
+using Base: @deprecate
 using AWS.AWSServices: s3
```
Comment (omus) [inline src/AWSS3.jl:40]: Correct ordering
```suggestion
using AWS.AWSServices: s3
using Base: @deprecate
```

**Comment by omus** [comment]
bors try

**Review by christopher-dG — APPROVED** [review]

**Pull request merged**